Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1168 @@
+        git_repo: Repo,
+        branch_metadata: Dict,
+        origin_base_metadata: Dict,
+    ):
+        super().__init__(
+            pack=pack,
+            pr=pr,
+            git_repo=git_repo,
+            branch_metadata=branch_metadata,
+            origin_base_metadata=origin_base_metadata,
+        )
+
+    def generate_skip_reason(self, not_allowed_changed_keys, **kwargs) -> str:      # type: ignore[override]
+        """
+        Args:
+            not_allowed_changed_keys: pack_metadata keys that was changed.
+        Returns: Reason why the condition failed, and pr skipped.
+        """
+        return SkipReason.ONLY_VERSION_CHANGED.format(
+            self.pack, not_allowed_changed_keys
+        )
+
+    def _check(
+        self, previous_result: Optional[ConditionResult] = None, **kwargs
+    ) -> ConditionResult:
+        """Checks if other pack metadata fields changed except ALLOWED CHANGED PACKS.
+        Args:
+            previous_result: previous check artifacts.
+
+        Returns(ConditionResult): whether the condition check pass,
+            or we should skip this pr from auto-bumping its release notes, with the reason why to skip.
+        """
+        not_allowed_changed_keys = [
+            k
+            for k, v in self.branch_metadata.items()
+            if k not in self.ALLOWED_CHANGED_KEYS
+            and self.origin_base_metadata.get(k) != v
+        ]
+        if not_allowed_changed_keys:
+            return ConditionResult(
+                should_skip=True,
+                reason=self.generate_skip_reason(
+                    not_allowed_changed_keys=not_allowed_changed_keys,
+                ),
+            )
+        else:
+            return ConditionResult(should_skip=False)
+
+
+class OnlyOneRNPerPackCondition(MetadataCondition):
+    def generate_skip_reason(self, rn_files: list, **kwargs) -> str:    # type: ignore[override]
+        """
+        Args:
+            rn_files: release notes files for the pack in current pr.
+        Returns: Reason why the condition failed, and pr skipped.
+        """
+        return SkipReason.MORE_THAN_ONE_RN.format(self.pack, [str(f) for f in rn_files])
+
+    def _check(
+        self, previous_result: Optional[ConditionResult] = None, **kwargs
+    ) -> ConditionResult:
+        """Checks that only one release notes files per pack was added.
+        Args:
+            previous_result: previous check artifacts.
+
+        Returns(ConditionResult): whether the condition check pass,
+            or we should skip this pr from auto-bumping its release notes, with the reason why to skip.
+        """
+        pack_new_rn_files = [
+            Path(f.filename)
+            for f in self.pr.get_files()
+            if f.status == "added"
+            and RELEASE_NOTES_DIR in Path(f.filename).parts
+            and self.pack in Path(f.filename).parts
+            and "md" in Path(f.filename).suffix
+        ]
+        if len(pack_new_rn_files) != 1:
```

Comment:
It is not legit to add more rn files to the pr. If it would happen I want to skip all processes. It should not happen, but if it happened by mistake I don't want to deal with it.